docs/tutorial/tutorial-8.rst

```diff
@@ -1,18 +1,150 @@
+opportunity exists to change context to another co-routine. The `asyncio
+<https://docs.python.org/3/library/asyncio.html>`__ module provides some other
+useful tools and primitives for asynchronous coding.
+
+Making the tutorial Asynchronous
+================================
+
+To make our tutorial asynchronous, modify the ``say_hello()`` event handler so
+it looks like this::
+
+    async def say_hello(self, widget):
+        if self.name_input.value:
+            name = self.name_input.value
+        else:
+            name = 'stranger'
+
+        async with httpx.AsyncClient() as client:
+            response = await client.get("https://jsonplaceholder.typicode.com/posts/42")
+
+        payload = response.json()
+
+        self.main_window.info_dialog(
+            "Hello, {}".format(name),
+            payload["body"],
+        )
+
+There are only 4 changes in this code from the previous version:
+
+1. The method is defined as ``async def``, rather than just ``def``. This tells
+   Python that the method is an asynchronous co-routine.
+
+2. The client that is created is an asynchronous ``AsyncClient()``, rather than a
+   synchronous ``Client()``. This tells ``httpx`` that it should operate in
+   asynchronous mode, rather than synchronous mode.
+
+3. The context manager used to create the client is marked as `async`. This tells
+   Python that there is an opportunity to release control as the context manager
+   is entered and exited.
+
+4. The `get` call is made with an `await` keyword. This instructs the app that
+   while we are waiting for the response from the network, the app can release control
+   to the event loop.
+
+Toga allows you to use regular methods or asynchronous co-routines as handlers;
+Toga manages everything behind the scenes to make sure the handler is invoked
+or awaited as required.
+
+If you save these changes and re-run the app (either with ``briefcase dev`` in
+development mode, or by updating and re-running the packaged app), there won't
+be any obvious changes to the app. However, when you click on the button to
+trigger the dialog, you may notice a number of subtle improvements:
+
+* The button returns to an "unclicked" state, rather than being stuck in a
+  "clicked" state.
+
+* The "beachball"/"hourglass" icon won't appear
+
+* If you move/resize the app window while waiting for the dialog to appear,
+  the window will redraw.
+
+* If you try to open an app menu, the menu will appear immediately.
 
 Next steps
 ==========
 
-We've now got an application with a custom icon on our desktop and phone! Is
-there anywhere else we can deploy a BeeWare app? Turn to :doc:`Tutorial 9
-<tutorial-9>` to find out...
+We've now got an application that is slick and responsive, even when it's
+waiting on a slow API. But it still looks like a tutorial app. Is there anything
+we can do about that? Turn to :doc:`Tutorial 9 <tutorial-9>` to find out...
```

Comment:
If pages 9 and 10 aren't ready yet, I think it would be better not to raise hopes and then disappoint them.

Reply:
My original thinking was that we could/should tease what was possible; however, you're probably right that it's better to stop the trail here until we actually fill out those tutorial pages (and, we should actually fill out those tutorial pages!)